app/src/main/java/io/github/notefydadm/notefy/database/DatabaseHandler.java

```diff
@@ -0,0 +1,178 @@
+            }
+
+        }else{
+            return null;
+        }
+
+    }
+
+    public static void userGetNoteListListener(final userGetNoteListListenerCallback callback, String userId){
+        FirebaseFirestore.getInstance().collection("notes").whereEqualTo("user_id",userId)
+                .addSnapshotListener(new EventListener<QuerySnapshot>() {
+                    @Override
+                    public void onEvent(@Nullable QuerySnapshot snapshots,
+                                        @Nullable FirebaseFirestoreException e) {
+                        if (e != null) {
+                            callback.onFailureOnListener(e);
+                            return;
+                        }else{
+                            for (DocumentChange dc : snapshots.getDocumentChanges()) {
+                                Note note = getNoteFromQueryDocumentSnapshot(dc.getDocument());
+                                switch (dc.getType()) {
+                                    case ADDED:
+                                        callback.onNoteAdded(note);
+                                        break;
+                                    case MODIFIED:
+                                        callback.onNoteModified(note);
+                                        break;
+                                    case REMOVED:
+                                        callback.onNoteRemoved(note);
+                                        break;
+                                }
+                            }
+
+                        }
+
+                    }
+                });
+    }
+
+
+
+    private static Note getNoteFromQueryDocumentSnapshot (QueryDocumentSnapshot snapshot){
+        String id = snapshot.getId();
+        String tittle = snapshot.getString("tittle");
+        String userId = snapshot.getString("user_id");
+        NoteState state = getStateFromString(snapshot.getString("state"));
+        boolean isFavourite = snapshot.getBoolean("is_favourite");
+        int colour = Math.toIntExact(snapshot.getLong("colour"));
+        LocalDateTime creationDate = LocalDateTime.ofEpochSecond(snapshot.getTimestamp("creation_date").getSeconds(),0, ZoneOffset.UTC);
+        LocalDateTime lastModifiedDate = LocalDateTime.ofEpochSecond(snapshot.getTimestamp("last_modified_date").getSeconds(),0, ZoneOffset.UTC);
+
+
+        Note item = new Note(id,tittle,userId,state,isFavourite,colour,creationDate,lastModifiedDate,null);
+        return item;
+    }
+
+    private static NoteState getStateFromString(String stateInString){
+        switch (stateInString){
+            case("DRAFT"):
+                return DRAFT;
+            case("PUBLISHED"):
+                return PUBLISHED;
+            case("HIDDEN"):
+                return HIDDEN;
+            case("ARCHIVED"):
+                return ARCHIVED;
+            case("DELETED"):
+                return DELETED;
+            default:
+                return null;
+
+        }
+
```

Comment:
Esto se puede simplificar bastante. Con una línea, de hecho.
```suggestion
        return NoteState.valueOf(stateInString);
```

Reply:
También se podría meter directamente donde lo uses en vez de tener un método aparte, pero eso ya va por gustos.
(De hecho, puede llegar a ser útil tener un método aparte, en según qué casos)